src/scenario.jl

```diff
@@ -329,19 +335,19 @@ function run_model(domain::Domain, param_set::DataFrameRow)::NamedTuple
     ps = NamedDimsArray(Vector(param_set); factors=names(param_set))
     return run_model(domain, ps)
 end
-function run_model(domain::Domain, param_set::NamedDimsArray)::NamedTuple
+function run_model(domain::Domain, param_set::YAXArray)::NamedTuple
     p = domain.coral_growth.ode_p
     corals = to_coral_spec(param_set)
     cache = setup_cache(domain)
 
     # Set random seed using intervention values
     # TODO: More robust way of getting intervention/criteria values
-    rnd_seed_val::Int64 = floor(Int64, sum(param_set(!=("RCP"))))  # select everything except RCP
+    rnd_seed_val::Int64 = floor(Int64, sum(param_set[Where(x -> x != "RCP")]))  # select everything except RCP
     Random.seed!(rnd_seed_val)
 
-    dhw_idx::Int64 = Int64(param_set("dhw_scenario"))
-    wave_idx::Int64 = Int64(param_set("wave_scenario"))
-    cyclone_mortality_idx::Int64 = Int64(param_set("cyclone_mortality_scenario"))
+    dhw_idx::Int64 = Int64(param_set[At("dhw_scenario")])
```

Comment:
With these `param_set[At(...)]`, I think `param_set.factor_name` should work and save a few keystrokes.

Reply:
I think that doesn't work. What I *can* do, is `param_set.factors` , becausa `:factors` is the name of an axis - but that doesn't return what we want here. The names of the factors are *labels* of that axis. I can think of two ways of selecting one single axis element in this case:

```
param_set[param_set.factors .== "dhw_scenario"].data[1]
param_set[At("dhw_scenario")]
```

![image](https://github.com/open-AIMS/ADRIA.jl/assets/8040719/2c100b1a-6a78-414c-a4e1-eb65c8863b2d)

The former is useful when we have more axes (I'm not sure if/how the later works with more axes).